### ISSUE TITLE
Change GPL-3.0 license to something more permissive 

### ISSUE THREAD

**Issue by joshdchang** [shahradelahi/node-vault#60]
### Describe the feature in detail (code, mocks, or screenshots encouraged)

This seems like a great library, and I would like to use it for an app I'm building independently. However, since I don't plan to distribute copies of my source code for my app, I can't use this library due to its license. I feel as though the GPL-3.0 license is too restrictive for a project such as this one. There are very few instances when someone would be using a vault in an open source project. A vault, by it's very nature, is supposed to be secret, and is a managed or self-hosted service deployment that costs something and therefore must make money in order to be sustainable. I understand the urge to encourage free and open software, but this license makes this software unusable to most who would want to use it legally. Thank you for considering!

### What type of pull request would this be?

None

**Comment by phisch** [comment]
Yeah, honestly, the license makes this project completely useless. It's used in sensitive projects, which you do not want to open source.

@shahradelahi MIT is the only way to go here.
